lambda/query-activity-log/tests/query-activity-log.test.ts

```diff
@@ -0,0 +1,292 @@
+import "aws-sdk-client-mock-jest";
+import { mockClient } from "aws-sdk-client-mock";
+import { SendMessageCommand, SQSClient } from "@aws-sdk/client-sqs";
+import { DynamoDBDocumentClient, QueryCommand } from "@aws-sdk/lib-dynamodb";
+
+import {
+  handler,
+  queryActivityLog,
+  sendSqsMessage,
+  validateTxmaEventBody,
+  validateUser,
+} from "../query-activity-log";
+import {
+  MUCKY_DYNAMODB_STREAM_EVENT,
+  TEST_DYNAMO_STREAM_EVENT,
+  TEST_TXMA_EVENT,
+  clientId,
+  eventType,
+  messageId,
+  queueUrl,
+  randomEventType,
+  sessionId,
+  tableName,
+  timestamp,
+  userId,
+} from "./test-helpers";
+import { Activity, ActivityLogEntry, UserActivityLog } from "../models";
+
+const dynamoMock = mockClient(DynamoDBDocumentClient);
+const sqsMock = mockClient(SQSClient);
+const activityList: Activity[] = [
+  {
+    type: "service_visited",
+    client_id: clientId,
+    timestamp,
+  },
+];
+
+const activityLogEntry: ActivityLogEntry = {
+  event_type: eventType,
+  session_id: sessionId,
+  user_id: userId,
+  timestamp,
+  activities: activityList,
+  truncated: true,
+};
+const userActivityLog: UserActivityLog = {
+  TxmaEvent: TEST_TXMA_EVENT,
+  ActivityLogEntry: activityLogEntry,
+};
+
+describe("queryActivityLog", () => {
+  beforeEach(() => {
+    dynamoMock.reset();
+
+    process.env.TABLE_NAME = tableName;
+
+    dynamoMock.on(QueryCommand).resolves({ Items: [activityLogEntry] });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Query activity log", async () => {
+    const activityLog: ActivityLogEntry | undefined = await queryActivityLog(
+      userId,
+      sessionId
+    );
+    expect(activityLog).not.toBeNull();
+    expect(activityLog).toEqual(activityLogEntry);
+  });
+
+  test("Query user service empty list", async () => {
+    dynamoMock.on(QueryCommand).resolves({ Items: undefined });
+    const activityLog: ActivityLogEntry | undefined = await queryActivityLog(
+      userId,
+      sessionId
+    );
+    expect(activityLog).not.toBeNull();
+    expect(activityLog).toBeUndefined();
+  });
+});
+describe("validateTxmaEventBody", () => {
+  test("doesn't throw an error with valid txma data", () => {
+    expect(validateTxmaEventBody(TEST_TXMA_EVENT)).toBe(undefined);
+  });
+  test("throws error when client_id is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            timestamp: new Date().toISOString,
+            event_name: "AUTH_AUTH_CODE_ISSUED",
+            user: {
+              user_id: userId,
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when timestamp is missing", () => {
```

Comment:
all the test files i've seen we have space above each test and each describe.  I think it helps readability a lot

Reply:
I assumed my prettier plugin would take care of it . I will add the space .